app/components/SingleProduct.js

```diff
@@ -0,0 +1,70 @@
+// - image
+// - title
+// - price
+// - description
+// - quantity selector
+// - add to cart (disabled if product is sold out)
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { fetchProduct } from '../redux/singleProduct'
+import { addCartItem } from '../redux/cart'
+
+class SingleProduct extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      quantity: 1,
+    }
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    this.props.setProduct(id)
+  }
+
+  handleChange(ev) {
+    const { name, value } = ev.target;
+    this.setState({ [name]: value })
+  }
+
+  render() {
+    const { selectedProduct, addCartItem, user, cart } = this.props;
+    return (
```

Comment:
Let's add a case for when the product couldn't be found (can just be a string "404 couldn't find product")

Reply:
sure. can you approve in the meantime and i'll add this in later?